Rare crash during mc mirror

## Expected behavior
No crash

## Actual behavior

```
panic: send on closed channel

goroutine 152 [running]:
github.com/minio/mc/cmd.(*ParallelManager).addWorker.func1(0xc0003ae960)
    /q/.q/sources/gopath/src/github.com/minio/mc/cmd/parallel-manager.go:80 +0x70
created by github.com/minio/mc/cmd.(*ParallelManager).addWorker
    /q/.q/sources/gopath/src/github.com/minio/mc/cmd/parallel-manager.go:69 +0x78
```

## Steps to reproduce the behavior
This happens when Minio server returns an error during listing,

You can fake the error using the following diff: 

```diff
diff --git a/cmd/fs-v1.go b/cmd/fs-v1.go
index 98e9cc60..5bb47b2f 100644
--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -1098,9 +1098,17 @@ func (fs *FSObjects) getObjectETag(ctx context.Context, bucket, entry string, lo
 	return extractETag(parseFSMetaMap(fsMetaBuf)), nil
 }
 
+var globalListObjects int
+
 // ListObjects - list all objects at prefix upto maxKeys., optionally delimited by '/'. Maintains the list pool
 // state for future re-entrant list requests.
 func (fs *FSObjects) ListObjects(ctx context.Context, bucket, prefix, marker, delimiter string, maxKeys int) (loi ListObjectsInfo, e error) {
+
+	globalListObjects++
+	if globalListObjects > 8 {
+		return loi, errInvalidArgument
+	}
+
 	if err := checkListObjsArgs(ctx, bucket, prefix, marker, delimiter, fs); err != nil {
 		return loi, err
 	}
```

Then mc mirror from source to target where source & target are two different aliases, the server that needs to have a faked error should be the source

## mc version
master

## System information
All

--- a/cmd/mirror-main.go
+++ b/cmd/mirror-main.go
@@ -516,8 +516,9 @@ func (mj *mirrorJob) startMirror(ctx context.Context, cancelMirror context.Cance
 				return
 			}
 			if sURLs.Error != nil {
+				stopParallel()
 				errCh <- sURLs.Error
-				continue
+				return
 			}
 
 			if sURLs.SourceContent != nil {

